### DISCUSSION TITLE
Welcome to constraintlayout Discussions!

### DISCUSSION THREAD

**Discussion by jafu888** [androidx/constraintlayout#424] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Post links to Articles, Videos, Tweets.

Topics:
* Clasic ConstraintLayout, ConstraintSets, How to achieve
* Clasics MotionLayout
* Compose ConstraintLayout
* Compose MotionLayout
* Link, feature request, etc.
* ConstraintLayout on other platforms (Swing, C++, etc.)

  To get started, comment below with an introduction of yourself and tell what you do with ConstraintLayout.

**Comment by rodrigomartind** [comment]
Hi everyone 👋, I'm [@rodrigomartind](https://twitter.com/RodrigoMartinD)
I'm a fan of ConstraintLayout & MotionLayout, I've contributed in [Carousel](https://developer.android.com/training/constraint-layout/motionlayout/carousel)  for MotionLayout (infinite carousel) and I have created the virtual layout [CircularFlow](https://developer.android.com/reference/androidx/constraintlayout/helper/widget/CircularFlow).  I also created examples of MotionLayout in Compose [here](https://github.com/androidx/constraintlayout/tree/main/projects/ComposeConstraintLayout) , and in XML [here](https://github.com/rodrigomartind/MixAnimationsMotionLayout).

**Comment by yangwulang** [comment]
Will it support Jetbrains Compose in the future？

**Reply by jafu888** [reply]
No decision has been made on that. It is a relatively small effort but would represent a break with the views world.

This year we are working to achieve feature parity (Views/Compose) of MotionLayout as well as polishing the Compose API.

Given the heavy use of Views in legacy it is critical we maintain support for ConstraintLayout/MotionLayout in Views.
Both Views and Compose versions share the same common core so that they do not diverge. 

Feature parity between Views and Compose helps teams that are considering migrating know that the can rebuild their ConstraintLayout/MotionLayout UI's in Compose. Also teams that are using both Views and Compose share a common core engine.
